### PR TITLE
Variables: enhance tidb_enable_gc_tuner

### LINKED ISSUE #40849: Enhance system variable `tidb_enable_gogc_tuner `
## Enhancement
The system variable `tidb_enable_gogc_tuner` controls to enable or disable `GOGC` tuner which tune `GOGC` automaticly. If we enbale GOGC tuner, tidb will call `runtime.ReadMemStats` to collect memory stats which tiggers `STW` and blocks the tidb-server instance. We hope that it doesn't call `runtime.ReadMemStats` if we disable `GOGC` tuner, but it doesn't, tidb still call `runtime.ReadMemStats`  when GOGC tunner is disabled, we should make it what as we hope.

### PR DESCRIPTION
**Pull request by TonsnakeLin**
Signed-off-by: TonsnakeLin <lpbgytong@163.com>

### What problem does this PR solve?

Issue Number: close #40849

Problem Summary:
See the issue #40849.

### What is changed and how it works?
tidb-server doesn't call `runtime.ReadMemStats` when GOGC tuner is disabled.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: domain/domain.go [hunk 1]
```diff
@@ -2110,7 +2111,9 @@ func (do *Domain) updateStatsWorker(ctx sessionctx.Context, owner owner.Manager)
 			}
 
 		case <-readMemTricker.C:
-			memory.ForceReadMemStats()
+			if gctuner.EnableGOGCTuner.Load() {
```
Comment (hawkingrei) [inline domain/domain.go:2114]: Not only GOGCTuner needs it, but All also need it to upgrade memstat to avoid frequence STW.

Comment (hawkingrei) [reply domain/domain.go:2114]: Now we read the memstat by ```memory.ReadMemStats()```

**Review by hawkingrei — APPROVED** [review]

**Comment by TonsnakeLin** [comment]
@ekexium @cfzjywxk

**Review by ekexium — APPROVED** [review]

**Comment by TonsnakeLin** [comment]
> so we need to collect the memory stats to help another component do anything.

What's the " another component " ? MemoryLimit Tuner?

**Pull request merged**